src/bridge/EthereumJSBridge.js

```diff
@@ -403,7 +403,10 @@ const EthereumBridge: WalletBridge<Transaction> = {
       ? Promise.resolve()
       : Promise.reject(new NotEnoughBalance()),
 
-  getTotalSpent: (a, t) => Promise.resolve(t.amount.plus(t.gasPrice.times(t.gasLimit))),
+  getTotalSpent: (a, t) =>
+    t.amount.isGreaterThan(0) && t.gasPrice.isGreaterThan(0) && t.gasLimit.isGreaterThan(0)
+      ? Promise.resolve(t.amount.plus(t.gasPrice.times(t.gasLimit)))
+      : Promise.resolve(BigNumber(0)),
```

Comment:
(note: this is to be consistent with Libcore implementation. the zero value is used to block the "NEXT" on the UI)